### PR TITLE
Update spec 

### PR DESCRIPTION
**Pull request by gsathya**
Based on conclusion in https://github.com/tc39/proposal-set-methods/issues/35#issuecomment-448423425

### CODE REVIEW

File: spec/set-prototype-is-superset-of.html [hunk 1]
```diff
@@ -12,8 +11,7 @@
         1. Let _next_ be IteratorStep(_iter_).
         1. If _next_ is *false*, return *true*.
         1. Let _nextValue_ be ? IteratorValue(_next_).
-        1. Let _has_ be ? Call(_hasCheck_, _set_, « e »).
+        1. Let _has_ be ? Call(_hasCheck_, _set_, « _nextValue_.[[Value]] »).
```
Comment (ljharb) [inline spec/set-prototype-is-superset-of.html]: Is the .[[Value]] needed here? Doesn’t the ? / ReturnIfAbrupt unwrap the completion record for you?

File: spec/set-prototype-is-disjoint-with.html [hunk 2]
```diff
@@ -3,17 +3,15 @@
...
     1. If IsCallable(_hasCheck_) is *false*, throw a *TypeError* exception.
-    1. For each _e_ that is an element of _entries_, do
-        1. Let _has_ be ? Call(_hasCheck_, _otherSet_, « e »)
+    1. Let _iter_ be ? GetIterator(_iterable_).
+    1. If _iter_ is undefined, throw a *TypeError* exception.
+    1. Repeat,
+        1. Let _next_ be IteratorStep(_iter_).
+        1. If _next_ is *false*, return *true*.
+        1. Let _nextValue_ be ? IteratorValue(_next_).
+        1. Let _has_ be ? Call(_hasCheck_, _set_, « e »).
```
Comment (ljharb) [inline spec/set-prototype-is-disjoint-with.html]: Not sure if this is a typo, but if so:
```suggestion
        1. Let _has_ be ? Call(_hasCheck_, _set_, « _nextValue_ »).
```

File: spec/set-prototype-is-superset-of.html (hunk 1 above)
Comment (gsathya) [reply spec/set-prototype-is-superset-of.html]: Fixed! I was mostly going off the Set constructor in the spec which was recently fixed too (https://github.com/tc39/ecma262/pull/1310) :smile: 

Updated all the other methods to do the same too.

File: spec/set-prototype-is-disjoint-with.html (hunk 2 above)
Comment (gsathya) [reply spec/set-prototype-is-disjoint-with.html]: Fixed.

**Pull request merged**